cmd/geth/dbcmd.go

```diff
@@ -759,3 +779,91 @@ func showMetaData(ctx *cli.Context) error {
 	table.Render()
 	return nil
 }
+
+func inspectHistory(ctx *cli.Context) error {
+	if ctx.NArg() == 0 || ctx.NArg() > 2 {
+		return fmt.Errorf("required arguments: %v", ctx.Command.ArgsUsage)
+	}
+	var (
+		address common.Address
+		slot    common.Hash
+	)
+	if err := address.UnmarshalText([]byte(ctx.Args().Get(0))); err != nil {
+		return err
+	}
+	if ctx.NArg() > 1 {
+		if err := slot.UnmarshalText([]byte(ctx.Args().Get(1))); err != nil {
+			return err
+		}
+	}
+	// Load the databases.
+	stack, _ := makeConfigNode(ctx)
+	defer stack.Close()
+
+	db := utils.MakeChainDatabase(ctx, stack, true)
+	defer db.Close()
+
+	triedb := utils.MakeTrieDatabase(ctx, db, false, false, false)
+	defer triedb.Close()
+
+	var (
+		err   error
+		start uint64 // the first history object to query
+		end   uint64 // the last history object to query (included)
+
+		// State histories are sorted by state ID rather than block number.
+		// To address this, load the corresponding block header and perform
+		// the conversion by this function.
+		blockToID = func(blockNumber uint64) (uint64, error) {
```

Comment:
I'd prefer if you move the `blockToID` out of the `var`-block, so things aren't that indented. Feel free to move these into the var-block though:
```
startNumber = ctx.Uint64("start")
endBlock  = ctx.Uint64("end")
stats *pathdb.HistoryStats
```